image_batch_resizer/img_batch_resizer.py

```diff
@@ -1,36 +1,44 @@
 import os
-
 from string_utils import remove_non_numerics
 from PIL import Image
+import shutil
 import multiprocessing
-# from get_configs import get_config
-# config = get_config("Botany_PIC")
-source_dir = "/admin/picturae_drive_mount/CAS_for_OCR"
+import time
+source_directory = "/storage_01/picturae/delivery"
+dest_directory = "/admin/picturae_drive_mount/CAS_for_OCR"
+tmp_folder = "/admin/web-asset-importer/image_batch_resizer/tmp_resize"
+
 
```

Comment:
this whole funciton is... deep. recommend refactoring now or at a later date. Chatgpt did this, and it's more or less what I"m thinking. Code not checked for validity: 
import os
import shutil
from PIL import Image

def convert_tiff_folder(source_dir, dest_dir, tmp_dir, min_bar, quality, max_size_kb, resize_to=None):
    """Convert TIFF images in a directory to resized JPEGs with specified constraints."""
    for root, dirs, files in os.walk(source_dir):
        if 'undatabased' in dirs:
            process_directory(root, dest_dir, tmp_dir, min_bar, quality, max_size_kb, resize_to)

def process_directory(root, dest_dir, tmp_dir, min_bar, quality, max_size_kb, resize_to):
    tiff_dir = os.path.join(root, 'undatabased')
    output_dir = os.path.join(dest_dir, os.path.basename(root), 'resized_jpg')
    os.makedirs(output_dir, exist_ok=True)
    
    for file_name in os.listdir(tiff_dir):
        if file_name.endswith(".txt"):
            continue
        
        barcode = int(remove_non_numerics(file_name))
        if barcode >= min_bar and "Cover" not in file_name:
            if file_name.lower().endswith(('.tiff', '.tif')):
                process_file(tiff_dir, file_name, output_dir, tmp_dir, quality, max_size_kb, resize_to)

def process_file(tiff_dir, file_name, output_dir, tmp_dir, quality, max_size_kb, resize_to):
    base_name = os.path.splitext(file_name)[0]
    output_file_name = f"{base_name}.jpg"
    output_file_path = os.path.join(output_dir, output_file_name)
    tmp_file_path = os.path.join(tmp_dir, output_file_name)

    if should_skip(output_file_path, max_size_kb):
        print(f"Skipping {file_name}, already processed.")
        return

    file_path = os.path.join(tiff_dir, file_name)
    img_quality = quality

    with Image.open(file_path) as image:
        print(f"resizing file {file_name}")

        if resize_to is not None:
            image = image.resize(resize_to)

        while img_quality > 20:
            image.save(tmp_file_path, 'JPEG', quality=img_quality, optimize=True, subsampling=0)
            current_size_kb = os.path.getsize(tmp_file_path) / 1024

            if current_size_kb <= max_size_kb:
                shutil.copyfile(tmp_file_path, output_file_path)
                os.remove(tmp_file_path)
                print(f"Image {file_name} resized successfully")
                break

            img_quality = adjust_quality(img_quality, current_size_kb, max_size_kb)

        if img_quality <= 20:
            print(f"Warning: Could not reduce {file_name} to under {max_size_kb}KB without dropping below min quality.")

def should_skip(output_file_path, max_size_kb):
    if os.path.exists(output_file_path):
        size = os.path.getsize(output_file_path)
        return size / 1024 <= max_size_kb
    return False

def adjust_quality(img_quality, current_size_kb, max_size_kb):
    if current_size_kb > (max_size_kb + 300):
        return img_quality - 5
    return img_quality - 1

def remove_non_numerics(s):
    return ''.join(filter(str.isdigit, s))